_posts/2020-03-27-simulating-a-pandemic-with-elixir.md

```diff
@@ -0,0 +1,149 @@
+provides a wrapper for managing its state. The process is initialized with a
+map containing a single key `:infected`. `Agent.update/2` (that is, a function
+taking 2 arguments) is then used to update the state of the process, and
+`Agent.get/2` is used to get the current state of the process. The `&` syntax
+is shorthand for lambdas, consider the Clojure equivalent `#(assoc % :infected
+true)`.
+
+Behind the scenes, `Agent.update/2` sends a message to the process created by
+start_link. In the process itself, the process receives the message and handles
+the update to its state. In both `update` and `get` the caller waits until the
+operation is complete, and thus the functions are synchronous.
+
+Scientific stuff
+----------------
+
+<img src="/img/simulating-pandemics-with-elixir/science.jpeg"
+     alt="A picture of a dog behing chemistry equipment with text 'I have no idea what I'm doing'" />
+
+I'm not a virologist, statistician nor do I have anything to do with
+healthcare. None of the numbers or formulas are based on real-life data. That
+said, my computer science background has taught me to use the
+[Stetson-Harrison method](https://www.urbandictionary.com/define.php?term=Stetson-Harrison%20method)
+for approximating various results, so that's what we're using here. The end
+result looks nice, and that's all that matters, right?
+
+```elixir
+{dx, dy} = distance(person_count, a, b)
+delta = :math.sqrt(:math.pow(dx, 2) + :math.pow(dy, 2))
+probability = (person_count / (100 * delta)) * infection_rate
+```
+
+Nodes will be placed in a grid. On every simulation tick, each infected person
+has a chance to infect each connected cell based on node distance. Each
+infected person also has a small chance (0.001%) to die on each tick. If there
+are enough infected persons at any given time (20% of the population) the death
+chance grows tenfold.
+
+Scaling up
+----------
+
+<figure>
+  <video width="600" height="248" controls="controls">
+  <source src="/img/simulating-pandemics-with-elixir/output-1.webm" type="video/webm"></source>
+  </video>
+  <figcaption style="text-align: center">Video: simulator ran with 400 nodes</figcaption>
+</figure>
+<br />
+
+How big graphs Elixir can handle without thinking too much about the
+implementation? I built a simple simulator that connects a bunch of nodes to
+each other, forming a weighted undirected graph with weights based on node
+distance. The weights are used for probabilities of the infection spreading. If
+infected, each node has a small chance to die on every tick. The probability of
+death increases if the network has too many simultaneously infected nodes. With
+10000 persons, the simulator runs without any optimizations at roughly 1 second
+per tick. Since the simulator runs for roughly 50 ticks before nobody is
+infected anymore, this is fast enough. Note that the processes are not executed
+concurrently: each process is sent a message, then that process is waited for a
```

Comment:
This might be clearer if you said what sends the message: "X sends each process a message and then waits for a reply before sending a message to the next process."